pydantic/tools.py

```diff
@@ -15,7 +21,7 @@ def _generate_parsing_type_name(type_: Any) -> str:
 
 
 @lru_cache(maxsize=2048)
```

Comment:
I wonder if this or other random caches we probably have might be messing with our references and causing those weak ref issues

Reply:
I don't think so, as this isn't used anywhere else.